base/stream.jl

```diff
@@ -764,19 +764,30 @@ julia> read(err, String)
 "stderr messages"
 ```
 
-See also `Base.link_pipe!`.
+See also [`Base.link_pipe!`](@ref).
 """
 Pipe() = Pipe(PipeEndpoint(), PipeEndpoint())
 pipe_reader(p::Pipe) = p.out
 pipe_writer(p::Pipe) = p.in
 
+"""
+    link_pipe!(pipe; reader_supports_async=false, writer_supports_async=false)
+
+Initialize `pipe` and link the `in` endpoint to the `out` endpoint. The keyword
+arguments `reader_supports_async`/`writer_supports_async` correspond to
+`OVERLAPPED` on Windows and `O_NONBLOCK` on POSIX systems. They should be `true`
+unless they'll be used by an external program (e.g. the output of a command
+executed with [`run`](@ref)).
+"""
 function link_pipe!(pipe::Pipe;
                     reader_supports_async = false,
                     writer_supports_async = false)
      link_pipe!(pipe.out, reader_supports_async, pipe.in, writer_supports_async)
      return pipe
 end
 
+public link_pipe!
```

Comment:
Seems like these should go in a centralized location like the export list, even if just at the top of a file, rather than in the middle?

Reply:
Yeah that's fine with me, I only put it inline because I saw that's how it was done for some other `public` symbols. How about in `exports.jl`?

Reply:
Oh I'm dumb, there already is a list in `exports.jl` :upside_down_face: I'll move it there.

Reply:
Moved in 807c7176f9, and I also took the liberty of moving the command for `remove_linenums!()` in 48627d0951.